cmd/csi-snapshotter/main.go

```diff
@@ -93,6 +93,7 @@ var (
 	metricsAddress              = flag.String("metrics-address", "", "(deprecated) The TCP network address where the prometheus metrics endpoint will listen (example: `:8080`). The default is empty string, which means metrics endpoint is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
 	httpEndpoint                = flag.String("http-endpoint", "", "The TCP network address where the HTTP server for diagnostics, including metrics and leader election health check, will listen (example: `:8080`). The default is empty string, which means the server is disabled. Only one of `--metrics-address` and `--http-endpoint` can be set.")
 	metricsPath                 = flag.String("metrics-path", "/metrics", "The HTTP path where prometheus metrics will be exposed. Default is `/metrics`.")
+	pollSnapshotPeriod          = flag.Duration("poll-snapshot-period", 10*time.Second, "Poll interval to check if a snapshot is ready. Default is 10 seconds.")
```

Comment:
Can you set the default to 0 so that drivers can opt-in to this new behavior?

Reply:
> Can you set the default to 0 so that drivers can opt-in to this new behavior?

I think changing the default is a benefit for end users here, so I would prefer to set it to a reasonable poll interval. We could increase the poll interval higher, if scalability is a concern. However if we do get a 429 from the APIServer, we'll encounter an error and then leverage exponential backoff.

I can make the default 0, and then refactor the logic to fallback on the exponential backoff when `pollSnapshotPeriod` is not set (eg: we need to adjust when `Forget()` is called if we do this).

@xing-yang What would you prefer (either keeping the logic as is but modifying the `pollSnapshotPeriod`, or refactoring to use the exponential rate limiting poll interval as a fallback).